KillTrack.lua

```diff
@@ -275,7 +275,7 @@ function KT.Events.ENCOUNTER_END(self, _, _, _, size)
     end
 end
 
-GameTooltip:HookScript("OnTooltipSetUnit", function(self)
+TooltipDataProcessor.AddTooltipPostCall(Enum.TooltipDataType.Unit,function(self, data)
```

Comment:
For the lints about undefined variables, they'll have to be added to the `.luacheckrc` file, to the first `read_globals` table (add `TooltipDataProcessor` and `Enum` in alphabetical order).

For the "unused argument `data`" lint, just replace `data` with `_` (or alternatively it just be removed since it's the last parameter, so it'll look like `function(self)` instead.

Since I've already got a review going on this line: Can you also add a space before `function`? :)

```suggestion
TooltipDataProcessor.AddTooltipPostCall(Enum.TooltipDataType.Unit, function(self, data)
```